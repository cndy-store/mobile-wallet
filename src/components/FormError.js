import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import { Icon, Text, View } from 'native-base';
import { styles as s } from 'react-native-style-tachyons';

class FormError extends React.Component {
  static propTypes = {
    message: PropTypes.string.isRequired
  };

  render() {
    return (
      <View style={styles.container}>
        <Icon style={s.pl3} name="warning" />
        <Text style={s.pl1}>{this.props.message}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start'
  }
});

export default FormError;
