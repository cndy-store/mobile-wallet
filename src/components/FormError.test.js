import React from 'react';
import { render } from '../__tests__/renderer';
import FormError from './FormError';

it('renders correctly', () => {
  const { toJSON } = render(<FormError message="Could not be saved!" />);
  expect(toJSON()).toMatchSnapshot();
});
