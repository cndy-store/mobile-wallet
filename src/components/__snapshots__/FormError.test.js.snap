// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
    ]
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#000",
            "fontSize": 30,
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    virtual={undefined}
  >
    ï‰§
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Could not be saved!
  </Text>
</View>
`;
